apps/web/src/views/Swap/V3Swap/containers/SwapCommitButton.tsx

```diff
@@ -142,22 +162,29 @@ export const SwapCommitButton = memo(function SwapCommitButton({
       return
     }
     if (!swapCallback) {
+      if (revertReason === 'insufficient allowance') {
+        setApprovalSubmitted(false)
+        setWallchainSecondaryStatus('found')
+        return
+      }
       return
     }
     setSwapState({ attemptingTxn: true, tradeToConfirm, swapErrorMessage: undefined, txHash: undefined })
     swapCallback()
       .then((res) => {
+        setWallchainSecondaryStatus('not-found')
         setSwapState({ attemptingTxn: false, tradeToConfirm, swapErrorMessage: undefined, txHash: res.hash })
       })
       .catch((error) => {
+        setWallchainSecondaryStatus('not-found')
         setSwapState({
           attemptingTxn: false,
           tradeToConfirm,
           swapErrorMessage: typeof error === 'string' ? error : error.message,
```

Comment:
```suggestion
          swapErrorMessage: typeof error === 'string' ? error : error?.message,
```

error not means Error type in catch clause, always use `?.` to retrive message